### PR TITLE
feat(bulk-load): sync bulk load from remote storage

### PR DESCRIPTION
**Pull request by hycdong**
This pull request is about sync bulk load info from remote storage while starting meta server.
- add `_sync_bulk_load_storage` to simplify code
- add code for sync bulk load info from remote storage
- add unit test

### CODE REVIEW

File: src/dist/replication/meta_server/meta_bulk_load_service.cpp [hunk 1]
```diff
@@ -20,14 +20,13 @@ bulk_load_service::bulk_load_service(meta_service *meta_svc, const std::string &
 void bulk_load_service::initialize_bulk_load_service()
 {
     task_tracker tracker;
-    error_code err = ERR_OK;
+    _sync_bulk_load_storage =
```
Comment (neverchanje) [inline src/dist/replication/meta_server/meta_bulk_load_service.cpp:23]: Is there any other initialization point of `_sync_bulk_load_storage`? Given that It's inited with a temporary tracker here, can you assure it's memory-safe for `_sync_bulk_load_storage` to be called in other functions?

File: src/dist/replication/meta_server/meta_bulk_load_service.cpp [hunk 2]
```diff
@@ -1215,39 +1214,85 @@ void bulk_load_service::on_control_bulk_load(control_bulk_load_rpc rpc)
...
+
+// ThreadPool: THREAD_POOL_META_STATE
+void bulk_load_service::sync_apps_bulk_load_from_remote_storage()
+{
+    std::string path = _bulk_load_root;
+    _sync_bulk_load_storage->get_children(
+        std::move(path), [this](bool flag, const std::vector<std::string> &children) {
+            if (flag && children.size() > 0) {
+                ddebug_f("There are {} apps need to sync bulk load status", children.size());
+                for (auto &elem : children) {
```
Comment (levy5307) [inline src/dist/replication/meta_server/meta_bulk_load_service.cpp:1236]: ``` for (const auto &elem : children) {```

File: src/dist/replication/meta_server/meta_bulk_load_service.h [hunk 3]
```diff
@@ -209,12 +209,20 @@ class bulk_load_service
...
+    void create_bulk_load_root_dir();
 
-    void sync_apps_bulk_load_from_remote_stroage(error_code &err, task_tracker &tracker);
+    void sync_apps_bulk_load_from_remote_storage();
+
+    void do_sync_app_bulk_load(int32_t app_id);
+
+    void sync_partitions_bulk_load_from_remote_storage(int32_t app_id, const std::string &app_name);
+
+    void do_sync_partition_bulk_load(const gpid &pid,
```
Comment (neverchanje) [inline src/dist/replication/meta_server/meta_bulk_load_service.h:220]: Most of your function names are designed very verbose. For example, it's weird for `bulk_load_service::on_sync_partition_bulk_load` to repeat the word 'bulk_load'. Naming it `bulk_load_service::on_sync_partition` doesn't make it less readable. You can refactor the name in the later PR if possible.

Comment (hycdong) [reply src/dist/replication/meta_server/meta_bulk_load_service.h:220]: I think it is okay for long function name if it is readable for who are reading the code. `on_sync_partition_bulk_load` is used to sync `partition_bulk_load_info` structure from remote storage, I will simplify those functions in further pull request.

Comment (neverchanje) [reply src/dist/replication/meta_server/meta_bulk_load_service.h:220]: I think longer function means more efforts to read them. It depends on case. If `on_sync_partition_bulk_load` was a function of `meta_service`, it's fine to give it a long name.

File: src/dist/replication/meta_server/meta_bulk_load_service.cpp [hunk 4]
```diff
@@ -1215,39 +1213,85 @@ void bulk_load_service::on_control_bulk_load(control_bulk_load_rpc rpc)
...
+// ThreadPool: THREAD_POOL_META_STATE
+void bulk_load_service::sync_apps_bulk_load_from_remote_storage()
+{
+    std::string path = _bulk_load_root;
+    _sync_bulk_load_storage->get_children(
+        std::move(path), [this](bool flag, const std::vector<std::string> &children) {
+            if (flag && children.size() > 0) {
+                ddebug_f("There are {} apps need to sync bulk load status", children.size());
+                for (const auto &elem : children) {
+                    uint32_t app_id = boost::lexical_cast<uint32_t>(elem);
```
Comment (acelyc111) [inline src/dist/replication/meta_server/meta_bulk_load_service.cpp:1236]: `boost::lexical_cast ` will throw exception when cast failed, are you sure the cast always success for `elem`?

File: src/dist/replication/meta_server/meta_bulk_load_service.cpp [hunk 5]
```diff
@@ -1215,39 +1213,85 @@ void bulk_load_service::on_control_bulk_load(control_bulk_load_rpc rpc)
...
 
-void bulk_load_service::sync_apps_bulk_load_from_remote_stroage(error_code &err,
-                                                                task_tracker &tracker)
+// ThreadPool: THREAD_POOL_META_STATE
+void bulk_load_service::do_sync_app_bulk_load(int32_t app_id)
 {
-    // TODO(heyuchen): TBD
+    std::string app_path = get_app_bulk_load_path(app_id);
+    _sync_bulk_load_storage->get_data(
+        std::move(app_path), [this, app_id, app_path](const blob &value) {
```
Comment (acelyc111) [inline src/dist/replication/meta_server/meta_bulk_load_service.cpp:1249]: `app_path ` is not needed to capture.

File: src/dist/replication/meta_server/meta_bulk_load_service.h (hunk 3 above)
Comment (hycdong) [reply src/dist/replication/meta_server/meta_bulk_load_service.h]: I have already updated function name.

File: src/dist/replication/meta_server/meta_bulk_load_service.cpp (hunk 4 above)
Comment (hycdong) [reply src/dist/replication/meta_server/meta_bulk_load_service.cpp]: Yes, `elem` is app_id

**Pull request merged**